src/go/k8s/internal/controller/redpanda/managed_decommission_controller.go

```diff
@@ -0,0 +1,580 @@
+	}
+
+	requestedReplicas, err := getRedpandaReplicas(ctx, c, rp)
+	if err != nil {
+		return err
+	}
+
+	adminAPI, err := buildAdminAPI(rp.GetHelmReleaseName(), rp.Namespace, int32(requestedReplicas), nil, valuesMap)
+	if err != nil {
+		return fmt.Errorf("creating AdminAPI: %w", err)
+	}
+
+	decomStatus, err := adminAPI.DecommissionBrokerStatus(ctx, decommissionNodeID)
+	if err != nil {
+		return fmt.Errorf("getting decommission broker status: %w", err)
+	}
+
+	if !decomStatus.Finished {
+		log.Info("decommission status not finished", "decommission-broker-status", decomStatus)
+		return &resources.RequeueAfterError{
+			RequeueAfter: wait.Jitter(decommissionWaitInterval, decommissionWaitJitterFactor),
+			Msg:          fmt.Sprintf("broker %d decommission status not finished", decommissionNodeID),
+		}
+	}
+
+	log.Info("Node decommissioned")
+	er.AnnotatedEventf(rp,
+		map[string]string{v1alpha1.GroupVersion.Group + revisionPath: rp.ResourceVersion},
+		corev1.EventTypeNormal, v1alpha1.EventTypeTrace, fmt.Sprintf("Node decommissioned: %d", decommissionNodeID))
+
+	err = podEvict(ctx, log, c, rp)
+	if err != nil {
+		return err
+	}
+
+	rp.Status.ManagedDecommissioningNode = nil
+
+	return updateRedpanda(ctx, rp, c)
+}
+
+var ErrZeroReplicas = errors.New("redpanda replicas is zero")
+
+func getRedpandaReplicas(ctx context.Context, c k8sclient.Client, rp *v1alpha1.Redpanda) (int, error) {
+	requestedReplicas := ptr.Deref(rp.Spec.ClusterSpec.Statefulset.Replicas, 0)
+	if requestedReplicas == 0 {
+		resourcesName := rp.Name
+		if rp.Spec.ClusterSpec.FullNameOverride != "" {
+			resourcesName = rp.Spec.ClusterSpec.FullNameOverride
+		}
+
+		var sts appsv1.StatefulSet
+		err := c.Get(ctx, types.NamespacedName{
+			Namespace: rp.Namespace,
+			Name:      resourcesName,
+		}, &sts)
+		if err != nil {
+			return 0, fmt.Errorf("getting statefulset : %w", err)
+		}
+
+		requestedReplicas = int(ptr.Deref(sts.Spec.Replicas, 0))
+	}
+
+	if requestedReplicas == 0 {
+		return 0, ErrZeroReplicas
+	}
+	return requestedReplicas, nil
+}
+
+func podEvict(ctx context.Context, l logr.Logger, c k8sclient.Client, rp *v1alpha1.Redpanda) error {
+	if rp.Status.ManagedDecommissioningNode == nil {
+		return nil
+	}
+
+	decommissionNodeID := int(*rp.Status.ManagedDecommissioningNode)
+	log := l.WithName("podEvict").WithValues("decommission-node-id", decommissionNodeID)
+
+	pod, err := getPodFromRedpandaNodeID(ctx, l, c, rp, decommissionNodeID)
+	if err != nil {
+		return fmt.Errorf("getting pod from node-id (%d): %w", decommissionNodeID, err)
+	}
+
+	log.Info("delete Pod PVC", "pod-name", pod.Name)
+	if err = utils.DeletePodPVCs(ctx, c, pod, log); err != nil {
+		return fmt.Errorf(`unable to remove VPCs for pod "%s/%s: %w"`, pod.GetNamespace(), pod.GetName(), err)
+	}
+	return c.Delete(ctx, pod)
+}
+
+func getPodFromRedpandaNodeID(ctx context.Context, l logr.Logger, c k8sclient.Client, rp *v1alpha1.Redpanda, nodeID int) (*corev1.Pod, error) {
```

Comment:
will this take care of the multiple rp in a single ns case?

Reply:
Yes as per `app.kubernetes.io/instance` label value:

```go
        labels.Set{
		"app.kubernetes.io/component": componentLabelValue,
		"app.kubernetes.io/instance":  rp.Name,
	}.AsSelector()
```

https://github.com/redpanda-data/redpanda-operator/pull/49/files#diff-3b50bc37a129716ec058fcced9e244f5a17a55f4d26f5d1919ebcde3773f9d28R163